File: python/PythagoreanTriplet.py
import math
def triplets_with_sum(number):
    N=float(number)
    triplets=[]
    for c in range(int(N/2)-1,int((math.sqrt(2)-1)*N),-1):
        D=math.sqrt(c**2-N**2+2*N*c)
        if D==int(D):
           triplets.append([int((N-c-D)/2),int((N-c+D)/2),c])

    return triplets

File: javascript/PythagoreanTriplet.js
export function triplets(options) {
    const min = options.minFactor || 2
    const max = options.maxFactor || options.sum;
    return [...Array(max - min).keys()].map(i => i + min)
        .map(a => {
            const b = Math.floor((options.sum * (options.sum - 2 * a)) / (2 * (options.sum - a)))
            const c = options.sum - a - b
            return c <= max ? new Triplet([a, b, c]) : undefined
        })
        .filter(triplet => triplet && triplet.isPythagorean())
}
class Triplet {
    constructor(triplet) {
        this.triplet = triplet
    }
    isPythagorean() {
        const [a, b, c] = this.triplet
        return a < b && b < c && a * a + b * b === c * c
    }
    toArray() {
        return this.triplet
    }
}
